serveradmin/resources/views.py

```diff
@@ -111,13 +110,15 @@ def index(request):
             hosts[server['hostname']] = dict(server)
 
     page = int(request.GET.get('page', 1))
```

Comment:
What will the paginator actually do if `page` is negative or 0?

Reply:
I did the following now:
You can not enter a invalid page number using the UI. If the user has manually submitted a invalid page with the Query string he probably wants something harmful. We raise a SuspiciousOperation exception now which leads to a 400 - Bad Request

Reply:
Never trust User Input. UI is nice but can always be bypassed. Changing html or just building the request on your own. So backend validation is the right thing to go. ;)